ambari-server/src/main/java/org/apache/ambari/server/topology/addservice/RequestValidator.java

```diff
@@ -194,9 +195,12 @@ void validateConfiguration() {
     }
 
     Configuration clusterConfig = getClusterDesiredConfigs();
-    clusterConfig.setParentConfiguration(state.getStack().getValidDefaultConfig());
+    if (request.getRecommendationStrategy().shouldUseStackDefaults()) {
+      clusterConfig.setParentConfiguration(state.getStack().getDefaultConfig());
```

Comment:
Stack defaults should be used even for `NEVER_APPLY`.  This is how it works for blueprints, too.  Otherwise one has to specify every config property explicitly.

The recommendation strategy is for stack advisor, not stack defaults.

Reply:
Agreed, I've also commented on this above.